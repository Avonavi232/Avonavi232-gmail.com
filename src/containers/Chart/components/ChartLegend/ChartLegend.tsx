import * as React from 'react';
import { chartLegend } from '../../contants';

import styles from './styles.module.css';

export const ChartLegend: React.FC = () => {
  return (
    <div className={styles['container']}>
      <h4 className={styles['title']}>Изменение цены</h4>
      <ul className={styles['legend']}>
        {chartLegend.map(item => (
          <li key={item.key} className={styles['legend-item']}>
            <span className={styles['legend-sign']} />
            <span>{item.name}</span>
          </li>
        ))}
      </ul>
    </div>
  );
};

ChartLegend.displayName = 'ChartLegend';
