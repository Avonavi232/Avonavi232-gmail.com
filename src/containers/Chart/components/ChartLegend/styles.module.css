.container {
  display: flex;
}

.title {
  font-weight: bold;
  font-size: 16px;
  line-height: 22px;
  margin: 0;
}

.legend {
  display: flex;
  align-items: center;
  color: rgba(21, 34, 66, 0.6);
  list-style-type: none;
  margin: 0 0 0 19px;
  padding: 0;
}

.legend-item {
  display: flex;
  align-items: center;
  font-size: 16px;
  line-height: 22px;
}

.legend-item:not(:last-child) {
  margin-right: 15px;
}

.legend-sign {
  width: 10px;
  height: 2px;
  margin-right: 7px;
}
