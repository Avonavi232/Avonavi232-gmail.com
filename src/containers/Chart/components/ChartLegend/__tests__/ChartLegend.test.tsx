import * as React from 'react';
import { shallow } from 'enzyme';

import { ChartLegend } from '../ChartLegend';

describe('ChartLegend', () => {
  it('должен рендериться без ошибок', () => {
    const wrapper = shallow(<ChartLegend />);
    expect(wrapper).toMatchSnapshot();
  });
});
