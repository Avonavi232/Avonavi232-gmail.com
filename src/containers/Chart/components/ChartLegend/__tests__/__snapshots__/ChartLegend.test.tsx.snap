// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartLegend должен рендериться без ошибок 1`] = `
<div
  className="src_shared_components_costestimate_v__components_chartlegend_styles_css__container"
>
  <h4
    className="src_shared_components_costestimate_v__components_chartlegend_styles_css__title"
  >
    Изменение цены
  </h4>
  <ul
    className="src_shared_components_costestimate_v__components_chartlegend_styles_css__legend"
  >
    <li
      className="src_shared_components_costestimate_v__components_chartlegend_styles_css__legend-item"
      key="offerPrice"
    >
      <span
        className="src_shared_components_costestimate_v__components_chartlegend_styles_css__legend-sign"
        style={
          Object {
            "backgroundColor": "#FF1F34",
          }
        }
      />
      <span>
        В объявлении
      </span>
    </li>
    <li
      className="src_shared_components_costestimate_v__components_chartlegend_styles_css__legend-item"
      key="estimatedPrice"
    >
      <span
        className="src_shared_components_costestimate_v__components_chartlegend_styles_css__legend-sign"
        style={
          Object {
            "backgroundColor": "#0468FF",
          }
        }
      />
      <span>
        Рыночная цена
      </span>
    </li>
  </ul>
</div>
`;
